staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -162,6 +171,9 @@ func NewRequest(c *RESTClient) *Request {
 		maxRetries:     10,
 		retryFn:        defaultRequestRetryFn,
 		warningHandler: c.warningHandler,
+
+		contentConfig:     contentConfig,
+		contentTypeNotSet: contentTypeNotSet,
```

Comment:
is this field used for something?

Reply:
I see, in subsequente commits

Reply:
Yeah, it tracks if the content type was manually set by the caller or if it was defaulted by not being set.  That is used in `UseProtobufAsDefault` so that a caller can use core clients but still force them to use JSON if they want.